File: Android_App_Dev - Java:Kotlin/LandmarksApp_ListView/app/src/main/java/com/muhammedalikocabey/java_deneme/MainActivity.java
package com.muhammedalikocabey.java_deneme;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import android.widget.Toast;

import com.muhammedalikocabey.java_deneme.databinding.ActivityMainBinding;

import java.util.ArrayList;
import java.util.stream.Collectors;


public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    ArrayList<Landmark> arrayList;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // setContentView(R.layout.activity_main);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);


        Landmark pisa = new Landmark("Pisa", "Italy", R.drawable.pisa);
        Landmark eiffel = new Landmark("Eiffel", "France", R.drawable.eiffel);
        Landmark colosseum = new Landmark("Colosseum", "Italy", R.drawable.colosseum);
        Landmark londonBridge = new Landmark("London Bridge", "United Kingdom", R.drawable.london_bridge);

        arrayList = new ArrayList<Landmark>();

        arrayList.add(pisa);
        arrayList.add(eiffel);
        arrayList.add(colosseum);
        arrayList.add(londonBridge);



        ArrayAdapter arrayAdapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1,
                arrayList.stream().map(landmark -> landmark.name).collect(Collectors.toList())
        );

        binding.listView.setAdapter(arrayAdapter);

        binding.listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent intent = new Intent(MainActivity.this, DetailsActivity.class);
                intent.putExtra("selected", arrayList.get(position));

                startActivity(intent);
            }
        });




    }









}

File: Android/TRY_LiveData_Kotlin/app/src/main/java/com/muhammedalikocabey/try_livedata_kotlin/MainActivity.kt
package com.muhammedalikocabey.try_livedata_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.muhammedalikocabey.try_livedata_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var mainViewModel: MainViewModel
    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        mainViewModel = ViewModelProvider(this@MainActivity).get(MainViewModel::class.java)

        mainViewModel.countLiveData.observe(this, Observer{
            binding.textView.text = "$it"
        })

        binding.button.setOnClickListener {
            mainViewModel.getCount()
        }

    }
}